portfolio/src/main/webapp/script.js

```diff
@@ -32,11 +32,106 @@ function addRandomFact() {
   factContainer.innerText = funFact;
 }
 
+/**
+ * Run functions on load
+ */
+function runOnLoadFunctions(){
+    getRandomGreeting();
+    getComments();
+}
+
 /**
  * Fetches random greeting from the server and adds it to the DOM.
  */
 function getRandomGreeting() {
     fetch('/greeting').then(response => response.text()).then((greeting) => {
     document.getElementById('greeting-header').innerText = greeting;
     });
+}
+
+/**
+ * Fetches comments from the server and adds it to the DOM.
+ */
+function getComments() {
+  fetch('/comments').then(response => response.json()).then((comments) => {
+    // stats is an object, not a string, so we have to
+    // reference its fields to create HTML content
+
+    const commentContainer = document.getElementById('comments-container');
+    createCommentElement(commentContainer, comments);
+  });
+}
+
+/** Creates a <div> element containing comment. */
+function createCommentElement(commentContainer, comments) {
+    if (comments.length == 0){
+        return commentContainer;
+    }
+
+    var i;
+    var commentDiv;
+    var paragraph;
+    var node;
+    var replyForm;
+    var formInput;
+    var commenter;
+
+    for (i = 0; i < comments.length; i++) {
+        commentDiv = document.createElement("div");
+        commentDiv.className = "comment-container";
+
+        paragraph = document.createElement("p");
+        paragraph.className = "commenter";
+        commenter = comments[i].commenter;
+        if (commenter === ""){
```

Comment:
You could try commenter.isEmpty() here

Reply:
This doensn't seem to work. 
`Uncaught (in promise) TypeError: commenter.isEmpty is not a function`

Reply:
Ah yes. Thought this was Java for a sec

Reply:
My bad! Sorry